src/irmin-http/swagger.yml

```diff
@@ -17,90 +17,193 @@ paths:
+          description: "branch already exists, no-op"
         201:
-          description: "" # TODO
+          description: "success"
   /branch/{branchName}:
+    parameters:
+    - in: "path"
+      name: "branchName"
+      description: "name of branch to operate on"
+      required: true
+      type: "string"
     get:
       summary: "get a branch by name"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
       responses:
         200:
-          description: "" #TODO
+          description: "success"
+          schema:
+            $ref: "#/definitions/Branch"
         404:
-          description: "" #TODO
+          description: "branch not found"
     delete:
       summary: "delete a branch by its ID"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
       responses:
         200:
-          description: "" #TODO
+          description: "success"
         404:
-          description: "" #TODO
+          description: "branch not found"
   /commits:
     get:
-      summary: "lists all commits"
+      summary: "lists all commits, newest to oldest (like `git log`)"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
       responses:
         200:
-          description: "" #TODO
-        404:
-          description: "" #TODO
+          description: "success"
+          schema:
+            type: array
+            items:
+              $ref: "#/definitions/Commit"
   /commits/{commitHash}:
+    parameters:
+    - in: "path"
+      name: "commitHash"
+      description: "hash of commit to operate on"
+      required: true
+      type: "string"
     get:
       summary: "get a commit by its hash"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
       responses:
         200:
-          description: "" #TODO
+          description: "success"
+          schema:
+            $ref: "#/definitions/Commit"
         404:
-          description: "" #TODO
-  /paths:
+          description: "commit not found"
+  /keys:
     get:
       summary: "gets all keys and their associated values (like `irmin tree`)"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
       responses:
         200:
-          description: "" #TODO
-        404:
-          description: "" #TODO
+          description: "success"
+          schema:
+            type: array
+            items:
+              $ref: "#/definitions/KV"
     post:
       summary: "add a new key/value pair to the store"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
+      parameters:
+      - in: body
+        name: body
+        description: "key/value pair to add"
+        required: true
+        schema:
+          $ref: "#/definitions/KV"
       responses:
         200:
-          description: "" #TODO
+          description: "key already exists, no-op"
```

Comment:
This was the conclusion I reached after trying to deciper the [webmachine state machine diagram](https://raw.githubusercontent.com/webmachine/webmachine/develop/docs/http-headers-status-v3.png). If you follow the diagram for a POST request, you can see that you end up at O20 ("Response includes an entity?"). Previously I had asked @yomimono about the definition of an entity and concluded that basically all REST API requests would contain an entity, so the decision diagram flows to O18 and then to "200 OK" as the final response code (assuming I also correctly understand what "multiple representations" means, which is far from guaranteed). If I was mistaken in my assumption that entities are in every request, then the correct response code is indeed 204 No Content. In any case, the diagram seems to indicate that some 2xx response code will be returned (even though I agree with yomimono that returning a success code in this case seems a little weird), but I need additional clarification on the above points before I can be certain which code is best.